components/applications-service/integration_test/suite_test.go

```diff
@@ -117,6 +118,16 @@ func (s *Suite) DeleteDataFromStorage() {
 
 // IngestService ingests a single HealthCheckEvent message into the database
 func (s *Suite) IngestService(event *habitat.HealthCheckEvent) {
+	eventsProcessed := s.Ingester.EventsProcessed()
```

Comment:
I don't think this is concurrency-safe -- can only one of these ever happen at the same time? If this could be called twice concurrently, we might miss an event.

Reply:
You are absolutely right! I can fix that for sure. 💯